### DISCUSSION TITLE
Installation schlägt fehl

### DISCUSSION THREAD

**Discussion by Computerboss3000** [thkl/hap-homematic#482] [General]
Servus zusammen,

auf Empfehlung der Community wolte ich heute das Plugin auf meiner CCU3 installieren, leider kam es während der Installation zu Fehlern und hoffe auf Eure Hilfe :)

ich habe jetzt zweimal versucht das Plugin hap-homematic zu installieren.

Nachdem ich die aktuellste Datei auf die CCU hochlade und auf installieren klicke erscheinen immer Fehlermeldungen.

Es heißt immer "Eine Komponente der Homematic Zentrale reagiert nicht mehr"

Anbei eine Auflistung der Komponenten die immer oben im Titel stehen:

HMIP-RF
Virtual Devices
BidCos-RF
ReGa

Nach einem Neustart sehe ich immer dass, dass Plugin bei Zusatzsoftware installiert wurde, jedoch sehe ich den neuen Punkt "Homekit" nicht in der Systemsteuerung. Auch über einstellen wird ein neuer Tab geöffnet, aber es kommt zu einem Fehler. Die Ports hatte ich bei Erstinsatllation freigegben aber sonst die Firewall ristrektiv auf blockieren und kein Zuriff für die APIs getsellt. Bei der 2 Installation habe ich alles auf Vollzugriff gestellt sowie alle Ports geöffnet. Beides mit dem gleichen Ergebnis.

Der durchführende Benutzer ist Administartor.

Ich kann mir leider nicht erklären was falsch läuft :(

Benötigt die CCU3 Internet weil das hat sie nicht, genauso wie der PC über den ich zugreife oder ist das Package auf Github bereits vollständig so das kein Internet benötigt wird?

Viele Grüße

**Comment by jb-home** [comment]
Die CCU braucht Internet.
Da im Installer nur ein Teil drin ist. Der Rest wird nachinstalliert.
Also die Kiste braucht Internet.

**Comment by Computerboss3000** [comment]
Vielen herzlichen Dank für die super schnelle und hilfreiche Antwort :)
Werde es gleich nochmal versuchen mit Internet für die CCU3.
Schön wäre es, wenn dies dabei stehen würde.

Ist es Ratsam die CCU3 nochmal neu aufzusetzen da ich beim deinstallieren auch einen Fehler angezeigt bekomme obwohl das zusatzpaket dann aber weg zu sein scheind, es wird zumindest nichts angezeigt. 
Oder ist dies nicht notwendig?

Viele Grüße

**Comment by jb-home** [comment]
Deinstallieren brauchst du nicht. Das versucht immer wieder die Daten herunterzuladen. Soweit ich das weiß.
Also nur Internet anschalten und ggf. die CCU dann rebooten.

**Comment by Computerboss3000** [comment]
Ah ok, hatte es dann immer wieder deinstalliert und da kam immer ein Fehler. Kann man einen Funktionstest der CCU3 machen ob durch die Installation/Deinstallation die Fehlerhaft bei mir war irgendwas in mitleidenschaft gezogen wurde sag ich mal?
Oder kann das garnicht passieren da das Grundsystem abgeschottet läuft?

Ich habe das Plugin nochmal installiert, es bleibt bei den Fehlern:
Es heißt immer "Eine Komponente der Homematic Zentrale reagiert nicht mehr"

Anbei eine Auflistung der Komponenten die immer oben im Titel stehen:

HMIP-RF
Virtual Devices
BidCos-RF
ReGa

Allerdings habe ich jetzt in der Systemsteurung den Homekit Button, super :)
Nunstellt sich mir noch die Frage ob ich das alles so belassen kann oder die CCU3 nochmal aufsetzen sollte, damit es sauber ist, da ich das Plugin ja oft deinstalliert und wieder installiert hatte?

Viele Grüße

**Comment by jb-home** [comment]
Also das installieren und deinstallieren hat sogeschen keinen Einfluß auf die CCU. Das passiert in einem separaten Verzeichnis und das Tool verbindet sich nur über die normalen Schnittstellen mit der CCU.
Somit sollte es keinen Einfluß auf die Komponenten haben.
Boote die Kiste 2 Mal durch. Wenn deine Komponenten dann immer noch Stress machen, dann liegt das Problem leider irgendwo anders. Da müsstest du dann die LOGs der CCU analysieren, warum die Komponenten ggf. abschmieren.

**Comment by jb-home** [comment]
Hast du noch ein anderes System am Laufen, welches sich gegen die CCU verbindet? Vielleicht iobroker oder ähnliches?

**Comment by Computerboss3000** [comment]
Vielen Dank!
Ich hatte mal Home-Assistant was zugegriffen hat, aber das tut es länger schon nicht mehr.
Hmm eine Verbindung zu HomeKit bekomme ich nicht hin und sobald ich das Internet bei der CCU3 wieder trenne wird mit garnichts mehr angezeigt beim Plugin, sehr schade :(

Andere Geräte, welche sich ebenfalls im extra IOT Netz befinden haben kein Problem in Homekit eingebunden zu werden, das scheint hier leider ein großes Problem zu sein.
Firewall hatte ich auch komplett deaktiviert sowie API vollzugriff aber leider weder ein hinzufügen über Homekit möglich, als eine funktionierende Oberfläche ohne Internet :(

**Comment by jb-home** [comment]
Hat die HAP Oberfläche denn mit Internet funktioniert?
Falls nicht, dann war HAP-HomeMatic ggf. noch nicht fertig mit der Installation. Das kann schon 15 Minuten oder länger dauern. Kommt auf die Hardware der CCU etc. an.
Generell sollte sich HAP-HomeMatic auch ohne Internet betreiben lassen. Da z.B. HAP-HomeMatic nur die Daten an das Apple TV, das iPad oder was auch immer übergibt. Dazu muss das Apple Gerät welches als HUB fungiert natürlich aus deinem IoT LAN erreichbar sein. Da du aber schon getrennte Netze hast, gehe ich davon aus, du weißt das ;)

**Comment by Computerboss3000** [comment]
Vielen Dank nochmal für die sehr schnelle Antwort :)
Mit Internet funktioniert die Oberfläche, ohne bekomme ich nur leere Bunte Boxen :(
Puh 15 Minuten hmm das heißt wenn die Weboberfläche erreichbar war, heißt es nicht das die Installation zuende war?
Vielleicht war ich da viel zu schnell dann :(

Genau dachte auch das es ohne Internet gehen müsste. Bei anderen Bridges klappt das auch wunderbar, habe da einiges im IOT Netz, das kann durch eine Firewallfreigabe perfekt auf mein Apple TV zugreifen, was im LAN hängt als Homekit HUB.
Ok während ich das hier schreibe sehe ich das jetzt ungefähr 1-2 Minuten nach öffnen der Homekit App alles geladen ist.
Oh man peinlich, da war ich viel zu ungeduldig :(

**Comment by jb-home** [comment]
TOP!
Dann kann das hier ja wieder zu.

Viel Spaß beim Einrichten :D

**Comment by Computerboss3000** [comment]
Ich habs jetzt nochmal deinstalliert und lass es gleich nochmal frishc installieren, da ich mit Homekit ja leider noch keine Verbindung zustande bekommen habe. Aber ich meld mich später nochmal obs geklappt hat :)

Vielen Dank dir nochmal an dieser Stelle!

**Comment by jb-home** [comment]
Wenn nur die bunten Boxen kommen, aber kein Inhalt, dann versuch mal den Cache zu leeren und probiere es erneut.
Wenn es dann immer noch nicht geht, schau mal mit F12 in die Konsole des Browsers, warum er was da nicht laden kann.

**Comment by Computerboss3000** [comment]
Hmm also auch mit Internet im anderen Netz kann ich leider die Bridge nicht hinzufügen :(
Die Instanzen werden mir direkt angezeigt in Homekit aber ich kann Sie nicht verbinden. Es wird sehr lange versucht und irgendwann steht dann in der Hom eApp "Gerät kann nicht hinzugefügt werden." Firewall ist aus und API Vollzugriff.
Kann HTTPS oder Authentifizierung in der CCU3 selbst eingetsellt ein Problem darstellen oder muss ich bei dem Plugin noch etwas konfigurieren?
Übrigends kam bei der Installation jetzt auch garkein Fehler oder ähnliches mehr :)
Weder die default Instanz noch die eigens angelegten lassen sich hinzufügen.
Rund 30 Minuten hatte ich dem Plugin zum installieren gegeben, denke das sollte passen.
Hmm wenn ich nur wüsste warum die Bridges sich nicht hinzufügen lassen. Browserdaten hatte ich auch komplett gelöscht.

**Comment by jb-home** [comment]
Hast du die Instanzen auch gepublisht?
https://github.com/thkl/hap-homematic/wiki/devices#publishing-settings

**Comment by Computerboss3000** [comment]
Hmm, also die Bridge müsste ich ja hinzufügen können oder?
Der Assistent sagt:
Perfekt! Schritt 2: Hier sind die neu angelegten Bridge Instanzen. Jetzt füge diese bitte zu HomeKit hinzu.
Aber da scheitert es.
Geräte hab ich jetzt noch keins hinzugefügt weil da steht das man das erst nach dem Birdge hinzufügen machen kann.
Ich hab jetzt mal den Haken gesetzt bei Publish für eine Bridge. Leider funktioniert es trotzdem nicht :(
Ich denke es gibt irgend ein Verbindungsproblem zwischen den Netzen. Wobei ja die Bridges wie gesagt direkt gefunden werden.
Auch die HAP UI funktioniert ohne internet ca 2-3 Minuten muss man warten dann ist diese aber geladen und sagt bereit zum verbinden mit HomeKit aber das klappt einfach nicht :( 
Auch den haken Kompatibilität zur Home App hatte ich rein gemacht sowie HTTPS aber nichts hat geholfen :( 
Firewall der CCU immernoch aus
Vielleicht hilft eine aktualisierung auf iOS 15 :)

**Comment by jb-home** [comment]
Du kannst Geräte auch zu der Instanz hinzufügen, bevor die Instanz mit HomeKit gekoppelt ist. Mach das einfach mal für mindestens 1 Gerät auf einer Instanz, die du gepublisht hast und versuche diese dann hinzuzufügen.
Wenn das dann immer noch nicht funktioniert, dann würde ich das Apple TV mal in das gleiche Netz wie die CCU packen, um Firewall Probleme erst mal auszuschließen.

**Comment by blackbasket** [comment]
Kannst du deine Firewall-Regeln einmal posten? Ist Multicast freigeschaltet?

**Comment by jb-home** [comment]
Ggf. hat HomeKit auch schon irgendwelche Sachen versucht anzugeben und ist jetzt in einem undefinierten Zustand.
Lösch auf der CCU mal in dem Ordner /usr/local/etc/config/addons/hap-homematic den Unterordner persistent.

**Comment by Computerboss3000** [comment]
Hmm Gerät habe ich mal hinzugefügt, hat leider nichts gebracht. 
Apple TV würde ich ungern dran rumbasteln, denn da laufen jetzt schon zu viele andere Sachen. Andere Gateways funktionieren wie gesagt ja auch....
Habe jetzt als letzten Versuch mich mal mit dem iPhone mich ins IOT Netz gehängt und von dort die Geräte Bridge versucht hinzuzufügen, das klappte auf anhieb sehr schnell und gut!
Jetzt scheint es zu funktionieren! Ich werde weiter testen und berichten, sollten sich negative Auffälligkeiten bemerkbar machen.
Aber erstmal scheint alles zu gehen, wäre wirklich klasse, wenn jetzt alles klappt :)

Vielen Dank nochmals an dieser Stelle!

**Comment by jb-home** [comment]
TOP!

Dann wird es irgendwo an der Firewall gelegen haben. Das ggf. eine Established/Related Verbindung nicht hergestellt werden konnte.
Das ist halt das Blöde an dem Apple Protokoll, dass da auch einiges über Multicast etc. geht. Falls du herausfindest an welcher Firewall Einstellung es lag, dann bitte gerne informieren...

**Comment by Computerboss3000** [comment]
Nochmal ein kleiner aktueller Stand, alles funktioniert super, es gab keine Probleme mehr.
Ich musste immer nur aus dem IOT Netz die Bridges hinzufügen zu Home Kit.
Leider kann ich nicht verstehen, warum es in dieser Richtung geht, da die Firewall Regeln in beide Richtungen die gleichen sind.
Deshalb hatte ich mich auch so gewundert, dass es von der einen Richtung aus klappt.

Viele Grüße

**Comment by tracer3575** [comment]
hi, auch ich habe wie schon Computerboss3000 versucht hat das plugin zu installieren. Mit der selben Fehlermeldung.
Komponenten reagieren nicht:

HMIP-RF
Virtual Devices
BidCos-RF
ReGa
Habe aber auch keinen Button in der Systemsteuerung mit "HomeKit"

Installation läuft nicht durch.
Was kann ich tun?

**Comment by jb-home** [comment]
@tracer3575  Darf die CCU ins Internet?

**Reply by tracer3575** [reply]
Ja, das sollte…
